### PR TITLE
move getOauth2Url() to an async task

### LINKED ISSUE #1426: App freezes during account setup
on 1.10
steps to reproduce:
1. fresh install
2. tap on 'Log in to your server'
3. enter email
4. focus on password field -> ANR

### PR DESCRIPTION
**Pull request by r10s**
fixes #1426

### CODE REVIEW

**Comment by r10s** [comment]
@cyBerta can you please try this pr with the phone that causes the freeze before?

File: src/org/thoughtcrime/securesms/util/task/ProgressDialogAsyncTask.java [hunk 1]
```diff
@@ -12,6 +16,8 @@
   private       ProgressDialog         progress;
   private final String                 title;
   private final String                 message;
+  private       boolean                cancellable;
```
Comment (link2xt) [inline src/org/thoughtcrime/securesms/util/task/ProgressDialogAsyncTask.java:19]: "cancellable" (two "l") is british. And "cancelable" is used in other places already.

File: src/org/thoughtcrime/securesms/util/task/ProgressDialogAsyncTask.java [hunk 2]
```diff
@@ -20,14 +26,21 @@ public ProgressDialogAsyncTask(Context context, String title, String message) {
     this.message          = message;
   }
 
+  public void setCancellable(@Nullable OnCancelListener onCancelListener) {
```
Comment (link2xt) [inline src/org/thoughtcrime/securesms/util/task/ProgressDialogAsyncTask.java:29]: Same here, think it is inconsistent with signal's code

File: src/org/thoughtcrime/securesms/util/task/ProgressDialogAsyncTask.java (hunk 1 above)
Comment (r10s) [reply src/org/thoughtcrime/securesms/util/task/ProgressDialogAsyncTask.java:19]: thanks, i'll fix that

**Pull request merged**